Tribler/Core/Modules/MetadataStore/store.py

```diff
@@ -177,6 +177,11 @@ def sqlite_disable_sync(_, connection):
             # Decay only happens while Tribler is running
             default_vsids.last_bump = datetime.utcnow()
 
+        # Flag to skip processing of personal channel torrent metadata. This is True by default because we assume
+        # the user always has the latest channel version and no need to update his own channel with gossipped data
+        # from the network. We need this flag in testing the processing of metadata.
+        self.skip_received_personal_metadata_payload = True
```

Comment:
When you use the parent object-level flag, you still leave yourself open to the possibility of getting old personal metadata during the moments this flag is true. Unless you make it an argument for mdblob processing functions, the following scenario can happen:
 1. You set `skip_received_personal_metadata_payload`  to `True` to restore your channel from disk;
 2. You call `process_channel_dir`. It starts processing mdblobs and eventually `yield`s to Twisted reactor.
 3. Someone from the network sends you your old metadata. This metadata gets processed by the GigaChannel community and because the `skip_received_personal_metadata_payload ` flag is `True`, this metadata is added to the database.
 4. Channel dir processing is resumed.

As a result, you've successfully restored your channel from the backup but also got some unexpected old entries from the network. T

Please, change the flag to be an argument.

Reply:
My line of thought here was if we're importing the user's channel, it should be synchronous or at least the channel should be locked so no extra payload can be processed.

Reply:
Good point. However, this will require additional effort (special starting mode, stopping IPv8, etc). And the proper usage is totally not obvious to anyone who did not read this thread. It is much safer and convenient to make it an argument (though the code becomes a bit uglier).